lib/fog/vsphere/requests/compute/create_vm.rb

```diff
@@ -166,6 +169,10 @@ def boot_options(attributes, vm_cfg)
             options[:bootRetryDelay]   = attributes[:boot_retry]
           end
 
+          if attributes[:secure_boot]
+            options[:efiSecureBootEnabled] = true
+          end
```

Comment:
This won't work if `boot_order` or `boot_retry` are not set. See https://github.com/fog/fog-vsphere/blob/62fb4b9b9b4e1071cd7fd6632244aa0896297a5e/lib/fog/vsphere/requests/compute/create_vm.rb#L25